### PR TITLE
Run worker in thread when register

### PR DESCRIPTION
**Pull request by tuhrig**

### CODE REVIEW

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGateway.kt [hunk 1]
```diff
@@ -23,6 +23,6 @@ class AwsKinesisInboundGateway(private val clientProvider: AwsKinesisClientProvi
 
         log.info("Running consumer to process stream [{}]...", handler.streamName())
 
-        worker.run()
+        Thread(worker).start()
```
Comment (jmatusewicz-bringmeister) [inline src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGateway.kt]: I would extract the thread creation to different class, so you can mock it in the unit test and do not need to sleep the thread.

File: src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGatewayTest.kt [hunk 2]
```diff
@@ -51,6 +53,6 @@ class AwsKinesisInboundGatewayTest : AbstractTest() {
 
         unit.register("foo-stream", { _: FooCreatedEvent, _: EventMetadata -> }, FooCreatedEvent::class.java, EventMetadata::class.java)
 
-        verify(worker).run()
+        verify(workerStarter).start(any())
```
Comment (jmatusewicz-bringmeister) [inline src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGatewayTest.kt]: why not verifying the call of start with the actual worker instance?

Comment (tuhrig) [reply src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisInboundGatewayTest.kt]: yeah, why not

**Comment by jmatusewicz-bringmeister** [comment]
👍

**Pull request merged**